httpclient/client.go

```diff
@@ -219,23 +232,52 @@ func (jc *HttpClient) ReadRemoteFile(downloadPath string, httpClientsDetails htt
 }
 
 func (jc *HttpClient) DownloadFile(downloadFileDetails *DownloadFileDetails, logMsgPrefix string, httpClientsDetails httputils.HttpClientDetails, retries int, isExplode bool) (*http.Response, error) {
-	resp, _, err := jc.downloadFile(downloadFileDetails, logMsgPrefix, true, httpClientsDetails, retries, isExplode)
+	resp, _, err := jc.downloadFileWithRetries(downloadFileDetails, logMsgPrefix, true, httpClientsDetails, retries, isExplode)
 	return resp, err
 }
 
 func (jc *HttpClient) DownloadFileNoRedirect(downloadPath, localPath, fileName string, httpClientsDetails httputils.HttpClientDetails) (*http.Response, string, error) {
```

Comment:
This method does not use retries. I think we should make that all download methods in the client use retries, with a default value of 3, unless configured otherwise.
I think with the uploads we're okay.

Reply:
Changed DownloadFileNoRedirect to uses the RetryExecutor.